portal/flatpak-portal.c

```diff
@@ -793,6 +793,9 @@ handle_spawn (PortalFlatpak         *object,
   gboolean devel;
   gboolean empty_app;
   g_autoptr(GString) env_string = g_string_new ("");
+  glnx_autofd int env_fd = -1;
+  const char *flatpak;
+  gboolean testing;
```

Comment:
```suggestion
  gboolean testing = FALSE;
```

If the `app_id` is not zero or `FLATPAK_PORTAL_MOCK_FLATPAK` is not set, i.e. we are not running the tests, we will end up using `testing` without initializing it first.

This can lead to `if (testing)` to evaluate positively even when it shouldn't.

Reply:
Oops, thanks - I tested this with the automated test but not "in real life".